crates/common/tedge_utils/src/file.rs

```diff
@@ -106,32 +113,13 @@ pub async fn move_file(
         }
     }
 
-    let original_permission_mode = match dest_path.is_file() {
-        true => {
-            let metadata = get_metadata(src_path)?;
-            let mode = metadata.permissions().mode();
-            Some(mode)
-        }
-        false => None,
-    };
+    // Copy source to destination including permission bits and remove source after successful copy operation
+    tokio::fs::copy(src_path, dest_path)
+        .and_then(|_| tokio::fs::remove_file(src_path))
+        .await?;
 
-    tokio::fs::rename(src_path, dest_path).await?;
```

Comment:
I would not directly copy/delete the file but try first to rename it as the copy/delete will not work if the file is larger than the free space (this is not unlikely with a firmware download on a resource-constraint device).

If would be better to fallback to a copy/delete only if the rename fails because of a [cross-filesystem rename](https://doc.rust-lang.org/nightly/std/io/enum.ErrorKind.html#variant.CrossesDevices), but unfortunately this is a nightly feature. So just try to rename and copy/delete in case of some error.

Reply:
Changed.